react/Geolocation.js

```diff
@@ -103,23 +128,30 @@ class Geolocation extends Component {
       rules: rules,
       address: address,
     })
+
     searchPickupAddressByGeolocationEvent({
       searchedAddressByGeolocation: true,
       confirmedGeolocation: true,
+      elapsedTime,
     })
   }
 
   getCurrentPositionError = error => {
-    this.setState({ isLoadingGeolocation: false })
     const { setAskForGeolocation, setGeolocationStatus } = this.props
+
+    this.setState({ isLoadingGeolocation: false })
+
+    const elapsedTime = Date.now() - this.startTime
+
     switch (error.code) {
       case 0: // UNKNOWN ERROR
         setAskForGeolocation(false)
         setGeolocationStatus(ERROR_COULD_NOT_GETLOCATION)
-        this.setCurrentActiveState(ERROR_COULD_NOT_GETLOCATION)
+        this.setCurrentActiveState(INITIAL)
```

Comment:
these state changes are the meat of this PR, really, along with the update in the error message